support pysigsci 3.10.0, add advanced rule copy

joeyleake opened the pull request:
- Support the latest pysigsci (3.10.0).
- Replaces the deprecated update_site_member with the new add_members_to_site.
- Uses the new copy_advanced_rule to copy advanced rules to new sites, removing the need for copy/paste emails to sigsci support.
- Catches an error in get_signal_rules caused when data was None. ie -{'totalCount': 0, 'data': None}
- Added warning for advanced rules with shortName containing a "+", which is currently unsupported

joeyleake committed:
support pysigsci 3.10.0, add advanced rule copy

- Support the latest pysigsci (3.10.0).
- Replaces the deprecated update_site_member with the new add_members_to_site.
- Uses the new copy_advanced_rule to copy advanced rules to new sites, removing the need for copy/paste emails to sigsci support.
- Catches an error in get_signal_rules caused when data was None. ie -{'totalCount': 0, 'data': None}
-

joeyleake committed:
remove commented out code

joeyleake committed:
fix typo

joeyleake committed:
update test

joeyleake committed:
Merge branch 'copy_advanced_rules' of https://github.com/turnerlabs/sigsci_site_manager into copy_advanced_rules

pizen commented on sigsci_site_manager/backup.py:
```diff
@@ -31,7 +31,10 @@ def get_custom_signals(api):
-    return filter_data(data['data'], keys)
+    try:
+        return filter_data(data['data'], keys)
+    except KeyError:
+        return {}, []
```
Add logic to `filter_data` so that it can't throw a `KeyError`, which will cover all instances of using `filter_data`

joeyleake replied:
Fixed

pizen commented on sigsci_site_manager/deploy.py:
```diff
@@ -112,12 +111,23 @@ def create_integrations(api, data):
     for item in data:
-        print('    %s (ID %s)' % (item['shortName'], item['id']))
+        if '+' in item['shortName']:
+            print("    Warning for %s: Rule names containing '+' not supported."
+                  % item['shortName'])
```
If there are advanced rules with '+' in the name it should probably still print out the email template for those specific rules so they can be copied. It should also `continue` on the loop so that unsupported rule names don't get a copy attempt that will fail, right?

joeyleake replied:
Changed it to just catch any failed copy and output an email request at the end.

pizen commented on sigsci_site_manager/merge.py:
```diff
@@ -296,31 +296,31 @@ def merge_integrations(api, data):
         if item['shortName'] not in rule_names:
-            email += '\n    %s (ID %s)' % (item['shortName'], item['id'])
-            make_request = True
+            if '+' in item['shortName']:
+                print("    Warning for %s: Rule names containing '+' not supported."
+                      % item['shortName'])
```
Same comments as above in the deploy code

joeyleake replied:
Changed it to just catch any failed copy and output an email request at the end.

joeyleake committed:
pr fixes, version bump

- revert the 'name contains +' warning
- now group all failed copies and output email request for manual copy
- fix the KeyError from the filter_data end
- bump version

joeyleake committed:
revert version
